### PR TITLE
fix: ui sdk unit tests

### PR DESCRIPTION
**Pull request by abhishek1508**
## Description

Fix UI SDK uncommented unit test cases.(Fixes [#3086](https://github.com/mapbox/mapbox-navigation-android/issues/3086))

- [x] I have added any issue links
- [x] I have added all related labels (`bug`, `feature`, `new API(s)`, `SEMVER`, etc.)
- [x] I have added the appropriate milestone and project boards

### Goal

Fix UI SDK uncommented unit test cases.

### Implementation

Fix UI SDK uncommented unit test cases.

## Testing

Please describe the manual tests that you ran to verify your changes

- [x] I have tested locally (including `SNAPSHOT` upstream dependencies if needed) through testapp/demo app and run all activities to avoid regressions
- [x] I have tested via a test drive, or a simulation/mock location app
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have updated the `CHANGELOG` including this PR

### CODE REVIEW

File: libnavigation-ui/src/test/java/com/mapbox/navigation/ui/NavigationViewModelTest.java [hunk 1]
```diff
@@ -11,40 +12,25 @@
...
-    NavigationViewRouter router = mock(NavigationViewRouter.class);
+    NavigationViewOptions mockNavigationViewOptions = mock(NavigationViewOptions.class);
     NavigationViewModel viewModel = new NavigationViewModel(application, navigation, mockedConnectivityController,
-            mapOfflineManager, router);
+        mapOfflineManager, mockNavigationViewOptions);
 
     viewModel.stopNavigation();
 
-    verify(navigation).removeMilestoneEventListener(null);
+    //verify(navigation).unregisterRouteProgressObserver(null);
```
Comment (Guardiola31337) [inline libnavigation-ui/src/test/java/com/mapbox/navigation/ui/NavigationViewModelTest.java:33]: If we comment this out, test is not checking anything 🤔 Am I missing something?

File: libnavigation-ui/src/test/java/com/mapbox/navigation/ui/NavigationViewModelTest.java [hunk 2]
```diff
@@ -85,48 +71,16 @@ public void updateRoute_navigationIsNotUpdatedWhenChangingConfigurations() {
...
     MapOfflineManager mapOfflineManager = mock(MapOfflineManager.class);
-    NavigationViewRouter router = mock(NavigationViewRouter.class);
+    NavigationViewOptions mockNavigationViewOptions = mock(NavigationViewOptions.class);
     NavigationViewModel viewModel = new NavigationViewModel(application, navigation, mockedConnectivityController,
-            mapOfflineManager, router);
+            mapOfflineManager, mockNavigationViewOptions);
     viewModel.onDestroy(true);
 
-    viewModel.updateRoute(route);
+    //viewModel.retrieveRoute().
```
Comment (Guardiola31337) [inline libnavigation-ui/src/test/java/com/mapbox/navigation/ui/NavigationViewModelTest.java:79]: What are we testing here?

File: libnavigation-ui/src/test/java/com/mapbox/navigation/ui/NavigationViewModelTest.java [hunk 3]
```diff
@@ -85,48 +71,16 @@ public void updateRoute_navigationIsNotUpdatedWhenChangingConfigurations() {
...
 
-    verify(navigation, times(0)).startNavigation(route);
-  }
-
-  @Test
-  public void navigationRouter_onDestroyInvokedWhenViewModelIsDestroyed() {
```
Comment (Guardiola31337) [inline libnavigation-ui/src/test/java/com/mapbox/navigation/ui/NavigationViewModelTest.java:99]: Is it safe to remove this test? I mean, is this scenario still covered by other test or was the functionality removed?

File: libnavigation-ui/src/test/java/com/mapbox/navigation/ui/NavigationViewModelTest.java [hunk 4]
```diff
@@ -85,48 +71,16 @@ public void updateRoute_navigationIsNotUpdatedWhenChangingConfigurations() {
...
-    NavigationViewRouter mockedRouter = mock(NavigationViewRouter.class);
-    NavigationViewModel viewModel = new NavigationViewModel(application, navigation, mockedConnectivityController,
-            mapOfflineManager, mockedRouter);
-    viewModel.onCleared();
-    verify(mockedRouter).onDestroy();
+    verify(navigation, times(0)).startTripSession();
   }
 
-  @Test
-  public void updateRoute_navigationIsUpdated() {
```
Comment (Guardiola31337) [inline libnavigation-ui/src/test/java/com/mapbox/navigation/ui/NavigationViewModelTest.java:112]: Is it safe to remove this test? I mean, is this scenario still covered by other test or was the functionality removed?

File: libnavigation-ui/src/test/java/com/mapbox/navigation/ui/map/MapFpsDelegateTest.java [hunk 5]
```diff
@@ -60,7 +75,7 @@ public void adjustFpsFor_thresholdSetWithCorrectManeuver() {
     when(batteryMonitor.isPluggedIn(any(Context.class))).thenReturn(false);
     MapFpsDelegate delegate = new MapFpsDelegate(mapView, batteryMonitor);
     RouteProgress routeProgress = buildRouteProgressWith("straight");
-    int maxFps = 5;
+    int maxFps = 30;
```
Comment (Guardiola31337) [inline libnavigation-ui/src/test/java/com/mapbox/navigation/ui/map/MapFpsDelegateTest.java:78]: Aren't we changing the test scenario here? Why was this needed `5` > `30`? Does the test fail with `5`?

File: libnavigation-ui/src/test/java/com/mapbox/navigation/ui/map/MapFpsDelegateTest.java [hunk 6]
```diff
@@ -75,7 +90,7 @@ public void adjustFpsFor_thresholdSetWithCorrectDuration() {
     when(batteryMonitor.isPluggedIn(any(Context.class))).thenReturn(false);
     MapFpsDelegate delegate = new MapFpsDelegate(mapView, batteryMonitor);
     RouteProgress routeProgress = buildRouteProgressWith(100d, 20d);
-    int maxFps = 5;
+    int maxFps = 30;
```
Comment (Guardiola31337) [inline libnavigation-ui/src/test/java/com/mapbox/navigation/ui/map/MapFpsDelegateTest.java:93]: Same as above re: Why was this needed `5` > `30`?

File: libnavigation-ui/src/test/java/com/mapbox/navigation/ui/map/MapFpsDelegateTest.java (hunk 5 above)
Comment (abhishek1508) [reply libnavigation-ui/src/test/java/com/mapbox/navigation/ui/map/MapFpsDelegateTest.java]: Made the change locally to test something but forgot to revert it back. Thanks for flagging. Fixed.

File: libnavigation-ui/src/test/java/com/mapbox/navigation/ui/map/MapFpsDelegateTest.java (hunk 6 above)
Comment (abhishek1508) [reply libnavigation-ui/src/test/java/com/mapbox/navigation/ui/map/MapFpsDelegateTest.java]: Made the change locally to test something but forgot to revert it back. Thanks for flagging. Fixed.

File: libnavigation-ui/src/test/java/com/mapbox/navigation/ui/NavigationViewModelTest.java (hunk 2 above)
Comment (abhishek1508) [reply libnavigation-ui/src/test/java/com/mapbox/navigation/ui/NavigationViewModelTest.java:79]: The functionality is removed as well. Forget to remove the test.
cc @Guardiola31337
